contrib/inventory/ec2.py

```diff
@@ -475,6 +479,13 @@ def connect_to_aws(self, module, region):
             connect_args['profile_name'] = self.boto_profile
             self.boto_fix_security_token_in_profile(connect_args)
 
+        if self.iam_role:
```

Comment:
It seemed more in line with the rest of the code to define the variable with a default value in `read_settings`.